app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,158 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  before_action :ensure_course_open, :ensure_not_registered, only: :create
+
+  def index #:nodoc:
+    @staff_requests = @registrations.staff
+    @student_requests = @registrations.student
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def create #:nodoc:
```

Comment:
Assignment Branch Condition size for create is too high. [24.17/15]<br>Method has too many lines. [17/10]